Library/Homebrew/linkage_checker.rb

```diff
@@ -90,6 +94,19 @@ def unexpected_broken_dylibs
     end
   end
 
+  def unexpected_present_dylibs
+    @unexpected_present_dylibs ||= @formula.class.allowed_missing_libraries.reject do |allowed_missing_lib|
```

Comment:
This will still fail, sorry, didn't think about that before. `allowed_missing_libraries` should perhaps return an empty Set or Array instead.

Reply:
> `allowed_missing_libraries` should perhaps return an empty Set or Array instead.

This is all this method does, so I'll just move its definition from `extend/os/linux/formula.rb` to `formula.rb`.